src/hooks/useShowHide.ts

```diff
@@ -1,11 +1,23 @@
 import { useCallback, useState } from "react";
 
-export const useShowHide = (isShownByDefault: boolean = false) => {
+interface UseShowHideProps {
+  isShownByDefault?: boolean;
+  targetClosest?: string[];
+}
+
+export const useShowHide = ({
+  isShownByDefault = false,
+  targetClosest,
+}: UseShowHideProps) => {
   const [isShown, setShown] = useState<boolean>(isShownByDefault);
 
-  const show = useCallback(() => {
-    setShown(true);
-  }, []);
+  const show = useCallback(
+    (e?) => {
+      if (targetClosest && e?.target.closest(targetClosest)) return;
+      setShown(true);
+    },
+    [targetClosest]
+  );
```

Comment:
It looks a bit hacky to my eyes. 
I see your implemented fix this way: we have some `targetClosest` exception. When the event was fired within that `targetClosest` selector we ignore the modal opening. Am I correct?

Let's consider another case. Let's say we want to make clickable not only the bookmark icon but the host in the event box in the schedule too.
![image](https://user-images.githubusercontent.com/63313732/122184196-00290980-ce5a-11eb-8b61-3d7e812249b9.png)
It means we have 2 clickable child elements. And we don't want the event modal to open when we click on them. And in fact, we can have more complex containers with more than 2 clickable child elements. Your implement version of the hook can't handle that case.

You could suggest transforming `targetClosest` to the array of selectors. But I wouldn't support that. I suppose would be better to keep things as isolated as possible. The name of the hook implies a straightforward logic of having a boolean status: "is shown" / "is hidden". Adding feature-specific things would one day overcomplicate the hook and increase the possibility of a bug.

Although I like your idea of building a generic solution, currently I can't quickly come up with a better solution than explicitly add the exceptions to the functions where it's needed. 

@0xdevalias @mike-lvov would love to hear your thoughts. 🙃

Reply:
All sounds totally reasonable. Just wanted to mention that the current implemtation should be able to handle an array of clickable child elements. For example, here: https://github.com/sparkletown/sparkle/blob/4e9c93cfe4b8d5f6e31a26fac8cb0c0f6fe8b040/src/components/templates/PosterHall/components/PosterPreview/PosterPreview.tsx#L50

By passing a string array to `targetClosest` similar to the example linked above, it should achieve a similar functionality. 

More broadly though, you're absolutely right that a more elegent general solution would be possible here.

Reply:
Oh! Sorry, one other question: the real goal of this is to deal with the bug in the event modal. If there's a simple way to add in the specific exception in `ScheduleEvent.tsx` without needing to copy over all the code from `useShowHide`, that's seems totally sufficient for the time being. Happy to do that if you could point me in the simplest direction to achieve that goal.

Reply:
I'd do something like this in the `ScheduleEvent.tsx`: 
```ts
  const onEventBoxClick = useCallback(
    (e) => {
      if (e.target.closest(".ScheduleEvent__bookmark")) return;

      showEventModal();
    },
    [showEventModal]
  );
  
  // some logic 
  
      <div className={containerClasses} onClick={onEventBoxClick}>
  // ...
```

Reply:
That was so much easier! Thanks! Cleaned up and much simpler now... really appreciate it the help